lang/values.md

```diff
@@ -215,6 +212,32 @@ Note in particular that `decode` ignores the bytes which are before, between, or
 `encode` in turn always and deterministically makes those bytes `Uninit`.
 (The [generic properties](#generic-properties) defined below make this the only possible choice for `encode`.)
 
+### Arrays
+
+```rust
+impl Type {
+    fn decode(Array { elem, count }: Self, bytes: List<AbstractByte>) -> Option<Value> {
+        if bytes.len() != elem.size() * count { throw!(); }
+        Value::Tuple(
+            bytes.chunks(elem.size())
+                .map(|elem_bytes| elem.decode(elem_bytes))
+                .try_collect()?,
+        )
+    }
+    fn encode(Array { elem, count }: Self, val: Value) -> List<AbstractByte> {
+        let Value::Tuple(values) = val else { panic!() };
+        assert_eq!(values.len(), count);
+        values.into_iter().flat_map(|value| {
+            let bytes = elem.encode(value);
+            assert_eq!(bytes.len(), elem.size());
+            bytes
+        }).collect()
+    }
+}
+```
+
+- TODO: Should we consider paddings between two adjacent elements in arrays?
```

Comment:
I don't think we need this remark, arrays are pretty tied down in how they are laid out. There is no room for padding here.